src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/FindAllReferencesHandler.cs

```diff
@@ -170,6 +170,11 @@ private async Task<VSReferenceItem[]> RemapReferenceItemsAsync(VSReferenceItem[]
                     continue;
                 }
 
+                // Temporary fix for codebehind leaking through
+                // Revert when https://github.com/dotnet/aspnetcore/issues/22512 is resolved
+                referenceItem.DefinitionText = FilterReferenceDisplayText(referenceItem.DefinitionText);
+                referenceItem.Text = FilterReferenceDisplayText(referenceItem.Text);
+
                 if (!RazorLSPConventions.IsRazorCSharpFile(referenceItem.Location.Uri))
```

Comment:
Do we need to keep the rest of the remapping logic in place, or can we safely remove it now that it's handled by Roslyn? 

FAR so C# only.